android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/EngineObserver.kt

```diff
@@ -46,6 +47,26 @@ internal class EngineObserver(
         store.dispatch(ContentAction.UpdateSearchTermsAction(tabId, ""))
     }
 
+    override fun onNavigateForward() {
+        store.dispatch(ContentAction.UpdateSearchTermsAction(tabId, ""))
+    }
+
+    override fun onGotoHistoryIndex() {
+        store.dispatch(ContentAction.UpdateSearchTermsAction(tabId, ""))
+    }
+
+    override fun onLoadData() {
+        store.dispatch(ContentAction.UpdateSearchTermsAction(tabId, ""))
+    }
+
+    override fun onLoadUrl() {
+        if (store.state.findTabOrCustomTab(tabId)?.content?.isSearch == true) {
```

Comment:
I was wondering where you decide to reset `isSearch`. I think this is fine for the use case here, but worth noting that if the user decides to refine the search in content (e.g., via typing and changing the search directly in the web page's search box) then we will not capture this next load as a search. 

However,that's already the case and the awesomebar won't have those search terms anyway. We do have logic to detect the search terms in this case and could address that in the future, if we ever wanted to: https://github.com/mozilla-mobile/firefox-android/blob/main/android-components/components/feature/search/src/main/java/mozilla/components/feature/search/ext/SearchEngine.kt#L101

Reply:
I believe this is the behavior described in the [first comment](https://github.com/mozilla-mobile/firefox-android/pull/124#issue-1440774722), in the last video. Should we file a follow-up for this? I think it would be worth investigating for search engine result pages which update the location based on the content displayed (see Reddit search engine), especially in the context of the Unified Search [tickets](https://mozilla-hub.atlassian.net/browse/FNXV2-22090).

Reply:
Filed [1804636](https://bugzilla.mozilla.org/show_bug.cgi?id=1804636) in this scope.